### PR TITLE
Issue 523

### PR DESCRIPTION
**Pull request by pcingola**
Added test case

Added tests in test_cli.py and test_client.py

Added testCors in test_views.py

Added testCors in test_views.py

Added testGetDataset in test_simulated_stack.py

Adding 404 check to testGetDataset

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Looks good, thanks @pcingola.

+1

File: ga4gh/frontend.py [hunk 1]
```diff
@@ -595,12 +595,14 @@ def oidcCallback():
...
-@app.route('/datasets/<no(search):id>')
+@DisplayedRoute(
+    '/datasets/<no(search):id>',
+    pathDisplay='/datasets/<id>')
 def getDataset(id):
-    raise exceptions.NotImplementedException()
+    return handleFlaskGetRequest(
+        id, flask.request, app.backend.runGetDataset)
 
+# The below paths have not yet been implemented
```
Comment (dcolligan) [inline ga4gh/frontend.py]: we can take out this comment, since there are no more methods in this file that have not been implemented

**Comment by dcolligan** [comment]
looks good aside from one nitpick

**Comment by pcingola** [comment]
Should I remove this comment and create a new PR squashing the new commit? 
Is there a better way that doesn't require a new PR? (I'm not fluent in Git).

**Comment by jeromekelleher** [comment]
The best way to do it is to use ``git rebase`` to squash the commits and then force-push your branch @pcingola. @dcolligan has a nice write up of the process [here](http://ga4gh-reference-implementation.readthedocs.org/en/develop/development.html#github-workflow).

**Comment by pcingola** [comment]
OK, I'll do that. Thank you for the pointer!

**Comment by pcingola** [comment]
Done, line removed.

**Comment by pcingola** [comment]
OK, I'm pretty sure that the method described in the docs doesn't work.
Any suggestions on how to squash commits? 
Or even better... Why squashing commits is required at all?

**Comment by jeromekelleher** [comment]
Squashing commits makes our lives easier when we're creating releases, and want to cherry pick certain commits onto the release branch. It also makes the history easier to understand, when we don't have lots of intermediate commits changing things back and forth. Squashing is a pain, and hurts the first few times you do it (it took me a _long_ time to get the hang of it) but it's totally worthwhile. Really, on a project like this, commit squashing is really useful. I'll revert back in a minute with a squash recipe for this PR.

**Comment by pcingola** [comment]
Created new PR #615 with squashed commits.
In my opinion, it's not only 'squshing' that is a pain, the whole Git model 
is broken (Git's motto should be "Committing code has never been so difficult").

**Comment by jeromekelleher** [comment]
OK, this branch is a bit of a mess now!
it reset --hard 6fa218820c2c45a9df9b48623ec1512c1268917b
$ git checkout get_dataset_by_id_squashed
$ git fetch upstream
$ git rebase -i upstream/develop

**Pull request closed without merging**